lib/Sema/ConstraintSystem.cpp

```diff
@@ -3521,10 +3521,26 @@ void ConstraintSystem::resolveOverload(ConstraintLocator *locator,
   if (isDebugMode()) {
     PrintOptions PO;
     PO.PrintTypesForDebugging = true;
-    llvm::errs().indent(solverState ? solverState->getCurrentIndent() : 2)
-      << "(overload set choice binding "
-      << boundType->getString(PO) << " := "
-      << adjustedRefType->getString(PO) << ")\n";
+
+    auto &log = llvm::errs();
+    log.indent(solverState ? solverState->getCurrentIndent() : 2);
+    log << "(overload set choice binding " << boundType->getString(PO)
+        << " := " << adjustedRefType->getString(PO);
+
+    auto openedAtLoc = getOpenedTypes(locator);
+    if (!openedAtLoc.empty()) {
+      log << " [";
+      llvm::interleave(
+          openedAtLoc.begin(), openedAtLoc.end(),
+          [&](OpenedType opened) {
+            opened.second->getImpl().getGenericParameter()->print(log);
```

Comment:
Should we forward `PO` to generic parameter as well? Not sure it would make any difference but is more a question =] 
```
opened.second->getImpl().getGenericParameter()->print(log, PO);
```

Reply:
I don't see why not! I've now updated all of the type printing in this `isDebugMode` block to use the same print function.

Reply:
It might be future proof but it’s really necessary at the moment because generic parameter cannot have anything that requires special printing options…